### PR TITLE
adds bug fix for cache location in case temp dir is defined by user

### PR DESCRIPTION
**Pull request by ankitaluthra1**
### Link to the issue in case of a bug fix.
b/320416020

### Testing details
1. Manual - NA
2. Unit tests - Yes
3. Integration tests - NA

### CODE REVIEW

File: internal/fs/fs.go [hunk 1]
```diff
@@ -231,7 +231,10 @@ func createFileCacheHandler(cfg *ServerConfig) (fileCacheHandler *file.CacheHand
 	if cacheLocation == "" {
 		if cfg.TempDir == "" {
 			cacheLocation = os.TempDir()
+		} else {
+			cacheLocation = cfg.TempDir
 		}
+
```
Comment (sethiay) [inline internal/fs/fs.go:237]: nit: remove this line ?

File: internal/fs/read_cache_test.go [hunk 2]
```diff
@@ -731,8 +732,13 @@ type FileCacheWithDefaultCacheLocation struct {
 	fsTest
 }
 
+type FileCacheWithUserDefinedTempAsCacheLocation struct {
+	fsTest
+}
+
 func init() {
 	RegisterTestSuite(&FileCacheWithDefaultCacheLocation{})
+	RegisterTestSuite(&FileCacheWithUserDefinedTempAsCacheLocation{})
```
Comment (ankitaluthra1) [inline internal/fs/read_cache_test.go:741]: @sethiay let me know if this needs to be registered in different init.. if its okay, should I add all register statements into one init method (better readable if they are all in one place)?

**Review by sethiay — APPROVED** [review]

File: internal/fs/read_cache_test.go (hunk 2 above)
Comment (sethiay) [reply internal/fs/read_cache_test.go:741]: Yes, adding at one place is better.

**Review by sethiay — COMMENTED** [review]
reverting approval for nit comment.

File: internal/fs/fs.go (hunk 1 above)
Comment (ankitaluthra1) [reply internal/fs/fs.go]: done

File: internal/fs/read_cache_test.go (hunk 2 above)
Comment (ankitaluthra1) [reply internal/fs/read_cache_test.go:736]: done

Comment (sethiay) [reply internal/fs/read_cache_test.go:736]: Thank you, nit: can we move init at the top - after constants definition and before first struct ?

Comment (ankitaluthra1) [reply internal/fs/read_cache_test.go]: done

**Review by sethiay — APPROVED** [review]

**Pull request merged**